Sources/SocksCore/Address.swift

```diff
@@ -167,3 +169,32 @@ extension ResolvedInternetAddress: CustomStringConvertible {
     }
 }
 
+extension Socket {
+    
+    public func remoteAddress() throws -> ResolvedInternetAddress {
+        var length = socklen_t(sizeof(sockaddr_storage))
+        let addr = UnsafeMutablePointer<sockaddr_storage>.init(allocatingCapacity: 1)
+        let addrSockAddr = UnsafeMutablePointer<sockaddr>(addr)
+        let res = getpeername(descriptor, addrSockAddr, &length)
+        guard res > -1 else {
+            addr.deallocateCapacity(1)
+            throw Error(.remoteAddressResolutionFailed)
+        }
+        let clientAddress = ResolvedInternetAddress(raw: addr)
+        return clientAddress
+    }
+    
+    public func localAddress() throws -> ResolvedInternetAddress {
+        var length = socklen_t(sizeof(sockaddr_storage))
+        let addr = UnsafeMutablePointer<sockaddr_storage>.init(allocatingCapacity: 1)
+        let addrSockAddr = UnsafeMutablePointer<sockaddr>(addr)
+        let res = getsockname(descriptor, addrSockAddr, &length)
+        guard res > -1 else {
+            addr.deallocateCapacity(1)
```

Comment:
Might be better to do a `defer` directly after the init. That way if someone adds a different early return, there won't be a bug.

```swift
let addr = UnsafeMutablePointer<sockaddr_storage>.init(allocatingCapacity: 1)
defer {
    addr.deallocateCapacity(1)
}
...
guard res > -1 else {
    throw Error(.localAddressResolutionFailed)
}
```